src/test/java/codeu/controller/AdminServletTest.java

```diff
@@ -0,0 +1,176 @@
+
+    adminServlet.setUserStore(mockUserStore);
+    adminServlet.setConversationStore(mockConvoStore);
+    adminServlet.setMessageStore(mockMessageStore);
+
+    Mockito.when(mockRequest.getRequestDispatcher("/WEB-INF/view/admin.jsp"))
+        .thenReturn(mockRequestDispatcher);
+  }
+
+  @Test
+  public void testDoGet() throws IOException, ServletException {
+    adminServlet.doGet(mockRequest, mockResponse);
+    mockRequest.getParameter("username");
+    Mockito.verify(mockRequestDispatcher).forward(mockRequest, mockResponse);
+  }
+
+  @Test
+  public void testGetStats() throws IOException, ServletException {
+    Mockito.when(mockRequest.getParameter("username")).thenReturn(null);
+
+    UUID fakeConversationId = UUID.randomUUID();
+    List<Conversation> fakeConversationList = new ArrayList<>();
+    Conversation fakeConversation =
+        new Conversation(fakeConversationId, UUID.randomUUID(), "test_conversation", Instant.now());
+    
+    fakeConversationList.add(fakeConversation);
+
+    Mockito.when(mockConvoStore.getAllConversations())
+        .thenReturn(fakeConversationList);
+
+    List<Message> fakeMessageList = new ArrayList<>();
+    fakeMessageList.add(
+        new Message(
+            UUID.randomUUID(),
+            fakeConversationId,
+            UUID.randomUUID(),
+            "test message",
+            Instant.now()));
+    Mockito.when(mockMessageStore.getAllMessages())
+        .thenReturn(fakeMessageList);
+
+    List totalUsers = mockUserStore.getUsers();
+    List totalMessages = fakeMessageList;
+    List totalConvos = mockConvoStore.getAllConversations();
+
+    Mockito.when(mockUserStore.getUsers()).thenReturn(totalUsers);
+    Mockito.when(mockMessageStore.getAllMessages()).thenReturn(totalMessages);
+    Mockito.when(mockConvoStore.getAllConversations()).thenReturn(totalConvos);
+
+    int numTotalUsers = totalUsers.size();
+    int numTotalMessages = totalMessages.size();
+    int numTotalConvos = totalConvos.size(); 
+
+    HttpSession mockSession = Mockito.mock(HttpSession.class);
+    
+    Mockito.when(mockRequest.getSession()).thenReturn(mockSession);
+
+    adminServlet.doPost(mockRequest, mockResponse);
+
+    Mockito.verify(mockSession).setAttribute("numUsers", numTotalUsers);
+    Mockito.verify(mockSession).setAttribute("numMessages", numTotalMessages);
+    Mockito.verify(mockSession).setAttribute("numConvos", numTotalConvos);
+
+    Mockito.verify(mockResponse).sendRedirect("/admin");
+  }
+
+  @Test
+  public void testDoPost_addAdmin() throws IOException, ServletException {
+    Mockito.when(mockRequest.getParameter("username")).thenReturn("test username");
+
+    adminServlet.doPost(mockRequest, mockResponse);
+
+    Mockito.verify(mockRequest).setAttribute("success", "Admin successfully added");
+    Mockito.verify(mockRequestDispatcher).forward(mockRequest, mockResponse);
+  }
+
+  // @Test
+  // public void testDoPost_ExistingUser() throws IOException, ServletException {
```

Comment:
Why did you comment this out?

Reply:
I just commented out this test because I copy pasted the whole thing from LoginServletTest before I edited it. I just forgot to delete this last test, but its irrelevant.

Reply:
I would just delete it so the code looks cleaner

Reply:
Yeah I have it deleted on the branch im currently working on so once I push that later we'll be all set